<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>omniplot.chipseq API documentation</title>
<meta name="description" content="" />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>omniplot.chipseq</code></h1>
</header>
<section id="section-intro">
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="omniplot.chipseq.call_superenhancer"><code class="name flex">
<span>def <span class="ident">call_superenhancer</span></span>(<span>bigwig: str, peakfile: str, stitch=25000, tss_dist: int = 0, plot_signals=False, gff: str = '', closest_genes: bool = False, nearest_k: int = 1, go_analysis: bool = False, gonames: list = ['GO_Biological_Process_2021', 'Reactome_2022', 'WikiPathways_2019_Human'])</span>
</code></dt>
<dd>
<div class="desc"><p>Find super enhancers and plot an enhancer rank.
</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>bigwig</code></strong> :&ensp;<code>str</code></dt>
<dd>A bigwig file.</dd>
<dt><strong><code>peakfile</code></strong> :&ensp;<code>str</code></dt>
<dd>A peak file.</dd>
<dt><strong><code>tss_dist</code></strong> :&ensp;<code>int</code>, optional</dt>
<dd>A distance from transcriptional start sites. If this option is more than 0, you need provide gff option too.</dd>
<dt><strong><code>stitch</code></strong> :&ensp;<code>int</code>, optional</dt>
<dd>The maximum distance of enhancers to stitch together.</dd>
<dt><strong><code>plot_signals</code></strong> :&ensp;<code>bool</code>, optional</dt>
<dd>Drawing super enhancer regions +- 10kb</dd>
<dt><strong><code>gff</code></strong> :&ensp;<code>str</code>, optional <code>(required when tss_dist is provided)</code></dt>
<dd>A gff or gtf file that contains gene annotations. You need to chose a correct genome version of a gff/gtf file</dd>
<dt><strong><code>closest_genes</code></strong> :&ensp;<code>bool=False,</code></dt>
<dd>If this is True, closest genes of SE will be shown, instead of genomic regions. You can specify the number
of closest genes by "nearest_k" option.</dd>
<dt><strong><code>nearest_k</code></strong> :&ensp;<code>int</code>, optional</dt>
<dd>The number of closest genes to be included. Default: 1.</dd>
<dt><strong><code>go_analysis</code></strong> :&ensp;<code>bool</code>, optional</dt>
<dd>GSEA analysis based on the rank of SEs.</dd>
<dt><strong><code>gonames</code></strong> :&ensp;<code>list=["GO_Biological_Process_2021","Reactome_2022","WikiPathways_2019_Human"]</code></dt>
<dd>Gene sets for GSEA analysis.</dd>
</dl>
<h2 id="returns">Returns</h2>
<pre><code>dict {"ax": ax, "positions":pos, "signals":y,"nearest_genes":_nearestgenes}
</code></pre>
<h2 id="raises">Raises</h2>
<h2 id="notes">Notes</h2>
<h2 id="references">References</h2>
<h2 id="see-also">See Also</h2>
<h2 id="examples">Examples</h2></div>
</dd>
<dt id="omniplot.chipseq.plot_average"><code class="name flex">
<span>def <span class="ident">plot_average</span></span>(<span>files: dict, bed: Union[str, list], order: list = [], extend: int = 500, palette: str = 'coolwarm', binsize: int = 10, clustering: str = '', n_clusters: int = 5, orientaion=False)</span>
</code></dt>
<dd>
<div class="desc"><p>Plotting bigwig files centered at peak regions.
</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>files</code></strong> :&ensp;<code>dict</code></dt>
<dd>A dictionary whose keys are sample names and values are file names</dd>
<dt><strong><code>bed</code></strong> :&ensp;<code>str</code></dt>
<dd>A bed file name containing the list of genomic regions</dd>
<dt><strong><code>order</code></strong> :&ensp;<code>list</code></dt>
<dd>A sample order to plot. if a clustering option (except kmeans_all) is chosen, the first sample is the target of clustering analysis.</dd>
<dt><strong><code>binsize</code></strong> :&ensp;<code>int</code></dt>
<dd>A bin size to reduce the bigwig signal resolution.</dd>
<dt><strong><code>clustering</code></strong> :&ensp;<code>str</code></dt>
<dd>Availables: ["kmeans", "kmeans_all", "kmeans_auto"]</dd>
<dt><strong><code>n_clusters</code></strong> :&ensp;<code>int</code></dt>
<dd>the number of clusters for clustering analysis. if you chose kmeans_auto, this option will be ignored.</dd>
</dl>
<h2 id="returns">Returns</h2>
<pre><code>dict {"values":data,"potisons":pos,"labels":labels}
values: dictionary containing sample names as the keys and signals as the values
positions: genomic positions sorted by signals displayed in the plot
labels: list of clustering labels if kmean clustering was performed
</code></pre>
<h2 id="raises">Raises</h2>
<h2 id="notes">Notes</h2>
<h2 id="references">References</h2>
<h2 id="see-also">See Also</h2>
<h2 id="examples">Examples</h2></div>
</dd>
<dt id="omniplot.chipseq.plot_bigwig"><code class="name flex">
<span>def <span class="ident">plot_bigwig</span></span>(<span>files: dict, bed: Union[str, list], gff: str, step: int = 100, stack_regions: str = 'horizontal', highlight: Union[str, list] = [], highlightname='')</span>
</code></dt>
<dd>
<div class="desc"><p>Plotting bigwig files in specified genomic regions in the bed file.
</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>files</code></strong> :&ensp;<code>dict</code></dt>
<dd>A dictionary whose keys are sample names and values are file names</dd>
<dt><strong><code>bed</code></strong> :&ensp;<code>str</code></dt>
<dd>A bed file name containing the list of genomic regions</dd>
<dt><strong><code>gff</code></strong> :&ensp;<code>str</code></dt>
<dd>A gff3 file with a corresponding genome version</dd>
<dt><strong><code>step</code></strong> :&ensp;<code>int</code></dt>
<dd>A bin size to reduce the bigwig signal resolution.</dd>
<dt><strong><code>show</code></strong> :&ensp;<code>bool</code></dt>
<dd>Whether or not to show the figure.</dd>
</dl>
<h2 id="returns">Returns</h2>
<p>{"ax":axes,"values":mat,"genes":geneset,"positions":pos} : dict
A dictionary containing matplot axes, signal values, geneset, and positions.
Raises</p>
<hr>
<h2 id="notes">Notes</h2>
<h2 id="references">References</h2>
<h2 id="see-also">See Also</h2>
<h2 id="examples">Examples</h2></div>
</dd>
<dt id="omniplot.chipseq.plot_bigwig_correlation"><code class="name flex">
<span>def <span class="ident">plot_bigwig_correlation</span></span>(<span>files: dict, chrom: str = 'chr1', step: int = 1000, palette: str = 'coolwarm', figsize: list = [6, 6], annot: bool = True, clustermap_param: dict = {}, peakfile: str = '')</span>
</code></dt>
<dd>
<div class="desc"><p>Calculate pearson correlations and draw a heatmap for bigwig files.
</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>files</code></strong> :&ensp;<code>dict</code></dt>
<dd>A dictionary whose keys are sample names and values are file names</dd>
<dt><strong><code>chrom</code></strong> :&ensp;<code>str</code></dt>
<dd>A chromsome to cal</dd>
<dt><strong><code>gff</code></strong> :&ensp;<code>str</code></dt>
<dd>A gff3 file with a corresponding genome version</dd>
<dt><strong><code>step</code></strong> :&ensp;<code>int</code></dt>
<dd>A bin size to reduce the bigwig signal resolution.</dd>
<dt><strong><code>show</code></strong> :&ensp;<code>bool</code></dt>
<dd>Whether or not to show the figure.</dd>
</dl>
<h2 id="returns">Returns</h2>
<p>{"ax":axes,"values":mat,"genes":geneset,"positions":pos} : dict
A dictionary containing matplot axes, signal values, geneset, and positions.
Raises</p>
<hr>
<h2 id="notes">Notes</h2>
<h2 id="references">References</h2>
<h2 id="see-also">See Also</h2>
<h2 id="examples">Examples</h2></div>
</dd>
<dt id="omniplot.chipseq.plot_genebody"><code class="name flex">
<span>def <span class="ident">plot_genebody</span></span>(<span>files: dict, bed: Union[str, list], order: list = [], extend: int = 500, palette: str = 'coolwarm', binsize: int = 10, clustering: str = '', n_clusters: int = 5)</span>
</code></dt>
<dd>
<div class="desc"><p>Plotting bigwig files around gene bodies.
</p>
<h2 id="parameters">Parameters</h2>
<dl>
<dt><strong><code>files</code></strong> :&ensp;<code>dict</code></dt>
<dd>A dictionary whose keys are sample names and values are file names</dd>
<dt><strong><code>bed</code></strong> :&ensp;<code>str</code></dt>
<dd>A bed file name containing the list of genomic regions</dd>
<dt><strong><code>order</code></strong> :&ensp;<code>list</code></dt>
<dd>A sample order to plot. if a clustering option (except kmeans_all) is chosen, the first sample is the target of clustering analysis.</dd>
<dt><strong><code>binsize</code></strong> :&ensp;<code>int</code></dt>
<dd>A bin size to reduce the bigwig signal resolution.</dd>
<dt><strong><code>clustering</code></strong> :&ensp;<code>str</code></dt>
<dd>Availables: ["kmeans", "kmeans_all", "kmeans_auto"]</dd>
<dt><strong><code>n_clusters</code></strong> :&ensp;<code>int</code></dt>
<dd>the number of clusters for clustering analysis. if you chose kmeans_auto, this option will be ignored.</dd>
</dl>
<h2 id="returns">Returns</h2>
<pre><code>dict {"values":data,"potisons":pos,"labels":labels}
values: dictionary containing sample names as the keys and signals as the values
positions: genomic positions sorted by signals displayed in the plot
labels: list of clustering labels if kmean clustering was performed
</code></pre>
<h2 id="raises">Raises</h2>
<h2 id="notes">Notes</h2>
<h2 id="references">References</h2>
<h2 id="see-also">See Also</h2>
<h2 id="examples">Examples</h2></div>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="omniplot" href="index.html">omniplot</a></code></li>
</ul>
</li>
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="omniplot.chipseq.call_superenhancer" href="#omniplot.chipseq.call_superenhancer">call_superenhancer</a></code></li>
<li><code><a title="omniplot.chipseq.plot_average" href="#omniplot.chipseq.plot_average">plot_average</a></code></li>
<li><code><a title="omniplot.chipseq.plot_bigwig" href="#omniplot.chipseq.plot_bigwig">plot_bigwig</a></code></li>
<li><code><a title="omniplot.chipseq.plot_bigwig_correlation" href="#omniplot.chipseq.plot_bigwig_correlation">plot_bigwig_correlation</a></code></li>
<li><code><a title="omniplot.chipseq.plot_genebody" href="#omniplot.chipseq.plot_genebody">plot_genebody</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>